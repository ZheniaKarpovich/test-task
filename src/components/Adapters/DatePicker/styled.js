import styled from 'styled-components';

import ReactDatePicker from 'react-datepicker';

const DatePicker = styled(ReactDatePicker)`
  width: 236px;
  height: 31px;
  max-height: 31px;
  padding: 7px;
  box-sizing: border-box;
  line-height: 16px;
  font-size: 14px;
  background: white;
  border-radius: 7px;
  border: solid 1px #979797;
  outline: none;
`;

export default DatePicker;
