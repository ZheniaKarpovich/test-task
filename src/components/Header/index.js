import React from 'react';
import PropTypes from 'prop-types';

import Logo from './images/logo.svg';
import Wrapper from './styled';

const Header = ({ component: Component }) => (
  <Wrapper>
    <img src={Logo} alt="logo" />
    {
      Component && <Component />
    }
  </Wrapper>
);

Header.defaultProps = {
  component: null,
};

Header.propTypes = {
  component: PropTypes.func,
};

export default Header;
