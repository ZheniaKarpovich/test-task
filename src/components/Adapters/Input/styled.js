import styled from 'styled-components';

const Input = styled.input`
  width: 236px;
  height: 31px;
  max-height: 31px;
  padding: 7px;
  box-sizing: border-box;
  line-height: 16px;
  font-size: 14px;
  background: white;
  border-radius: 7px;
  border: solid 1px #979797;
`;

export default Input;
