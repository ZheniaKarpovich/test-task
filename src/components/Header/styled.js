import styled from 'styled-components';
import { Flex } from '@rebass/grid';

export const Wrapper = styled(Flex)`
  width: 100%;
  height: 116px;
  padding: 30px 30px 30px 37px;
  box-sizing: border-box;
  justify-content: space-between;
  align-items: center;
  background-color: #7ed321;

  @media (max-width: 420px) {
    height: 77px;
  }
`;

export const Img = styled.img`
  @media (max-width: 420px) {
    width: 98px;
    height: 37px;
  }
`;
