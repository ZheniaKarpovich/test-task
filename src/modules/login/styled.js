import styled from 'styled-components';

import ButtonComonent from 'components/Button';

export const Wrapper = styled.div`
  width: 100%;
  height: calc(100% - 116px);
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Content = styled.div`
  width: 503px;
  height: 379px;
  box-sizing: border-box;
  padding: 77px 0 36px 0;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  border-radius: 44px;
  background-color: #e990f9;

  @media (max-width: 420px) {
    background-color: inherit;
  }
`;

export const Button = styled(ButtonComonent)`
  @media (max-width: 420px) {
    color: #e990f9;
    border: solid 3px #e990f9;
  }
`;

export const Img = styled.img`
  #facebook-logo:hover path {
    fill: red;
  }
  @media (max-width: 420px) {
    svg g path{
      fill: #e990f9;
    }
  }
`;
