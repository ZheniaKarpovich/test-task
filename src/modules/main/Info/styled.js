import styled from 'styled-components';

export const Wrapper = styled.div`
  width: 100%;
  height: calc(100% - 116px);
  padding: 85px 25px 0 25px;
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  overflow: auto;

  @media (max-width: 420px) {
    padding: 25px 25px 0 25px;
  }
`;

export const Content = styled.div`
  max-width: 500px;
  display: flex;
  box-sizing: border-box;
  flex-direction: column;
  align-items: center;
`;

export const Item = styled.div`
  width: 100%;
  min-width: 300px;
  margin-bottom: 30px;
  font-size: 14px;
  line-height: 1.71;
`;

export const Title = styled.h2`
  color: #7ed321;
  align-self: flex-start;
`;
