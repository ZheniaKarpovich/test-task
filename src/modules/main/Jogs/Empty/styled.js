import styled from 'styled-components';

import Button from 'components/Button';

export const Img = styled.img`
  @media (max-width: 420px) {
    width: 85px;
    height: 86px;
  }
`;

export const Wrapper = styled.div`
  width: 100%;
  height: calc(100% - 116px);
  overflow: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const Title = styled.div`
  margin-top: 47px;
  color: #b0b0b0;
  font-size: 24px;
`;

export const AddButton = styled(Button)`
  width: 249px;
  margin-top: 141px;
  border: solid 3px #e990f9;
  color: #e990f9;
`;
