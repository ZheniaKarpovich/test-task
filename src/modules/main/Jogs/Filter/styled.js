import styled from 'styled-components';
import ReactDatePicker from 'react-datepicker';

export const Wrapper = styled.div`
  width: 100%;
  min-height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #eaeaea;
`;

export const Item = styled.div`
  display: flex;
  align-items: center;
  font-size: 13px;
  margin: 0 20px 0 20px;
`;

export const DatePicker = styled(ReactDatePicker)`
  width: 100px;
  height: 31px;
  max-height: 31px;
  margin-left: 15px;
  padding: 7px;
  box-sizing: border-box;
  line-height: 16px;
  font-size: 14px;
  background: white;
  border-radius: 11px;
  border: solid 1px #979797;
  outline: none;
`;
