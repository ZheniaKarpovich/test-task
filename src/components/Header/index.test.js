import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Logo from './images/logo.svg';
import Header from './index';
import * as S from './styled';

const defaultProps = {
  logo: 'path',
  component: () => (<div id="smth">compoennt</div>),
};

describe('Header component', () => {
  test('render', () => {
    const wrapper = shallow(<Header {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('custom path', () => {
    const { logo } = defaultProps;
    const wrapper = shallow(<Header {...defaultProps} />);
    const img = wrapper.find(S.Img);

    expect(img.prop('src')).toEqual(logo);
  });

  test('default path', () => {
    const wrapper = shallow(<Header />);
    const img = wrapper.find(S.Img);

    expect(img.prop('src')).toEqual(Logo);
  });

  test('without children', () => {
    const wrapper = shallow(<Header />);
    const img = wrapper.children();

    expect(img).toHaveLength(1);
  });

  test('with children', () => {
    const wrapper = shallow(<Header {...defaultProps} />);
    const img = wrapper.children();

    expect(img).toHaveLength(2);
  });
});
