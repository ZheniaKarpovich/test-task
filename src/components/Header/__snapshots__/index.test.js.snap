// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    component={[Function]}
    logo="path"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          alt="logo"
          src="path"
        />,
        <component />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "logo",
          "src": "path",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  @media (max-width: 420px) {
    width: 98px;
    height: 37px;
  }
",
            ],
          },
          "displayName": "styled.img",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "box-sizing: border-box;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "display: flex;",
          [Function],
          [Function],
          [Function],
          [Function],
          "
  width: 100%;
  height: 116px;
  padding: 30px 30px 30px 37px;
  box-sizing: border-box;
  justify-content: space-between;
  align-items: center;
  background-color: #7ed321;

  @media (max-width: 420px) {
    height: 77px;
  }
",
        ],
      },
      "displayName": "Styled(Flex)",
      "foldedComponentIds": Array [
        "sc-bdVaJa",
        "sc-bwzfXH",
      ],
      "propTypes": Object {
        "alignItems": [Function],
        "flexDirection": [Function],
        "flexWrap": [Function],
        "justifyContent": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            alt="logo"
            src="path"
          />,
          <component />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "logo",
            "src": "path",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  @media (max-width: 420px) {
    width: 98px;
    height: 37px;
  }
",
              ],
            },
            "displayName": "styled.img",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "rules": Array [
            "box-sizing: border-box;",
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            "display: flex;",
            [Function],
            [Function],
            [Function],
            [Function],
            "
  width: 100%;
  height: 116px;
  padding: 30px 30px 30px 37px;
  box-sizing: border-box;
  justify-content: space-between;
  align-items: center;
  background-color: #7ed321;

  @media (max-width: 420px) {
    height: 77px;
  }
",
          ],
        },
        "displayName": "Styled(Flex)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
          "sc-bwzfXH",
        ],
        "propTypes": Object {
          "alignItems": [Function],
          "flexDirection": [Function],
          "flexWrap": [Function],
          "justifyContent": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
